Dataset class for BrainForm by Romani et al.

BRomans opened the pull request:
Implementation to load the BrainForm dataset from BIDS files.
Works with both locally stored files if path is provided, otherwise fetches the dataset from Zenodo.
Local files can also be newer sessions collected with the BrainForm neurogame and converted to BIDS format.

gcattan commented on moabb/datasets/romani_bf2025_erp.py:
```diff
@@ -0,0 +1,577 @@
+            events = events[np.argsort(events[:, 0])]  # sort by onset
+
+            # --- Infer number of unique targets automatically ---
+            # Assume event codes: "Target" and "NonTarget"
```
Shall we may be document this assumption in the description of the class?

BRomans replied:
hey thanks for reviewing and for pointing out that example, it helped me find and fix some bug.
Could you please provide me with an example on how summary_p300.csv should look like?
However I am still trying to understand why it finds some infs or NaN values, I am pretty sure they shouldn't be there.

```
Traceback (most recent call last):
  File "C:\Users\miche\Desktop\projects\moabb\examples\paradigm_examples\plot_within_session_p300.py", line 93, in <module>
    results = evaluation.process(pipelines)
  File "C:\Users\miche\Desktop\projects\moabb\moabb\evaluations\base.py", line 229, in process
    parallel_results = Parallel(n_jobs=self.n_jobs)(
  File "C:\Users\miche\Desktop\projects\moabb\.venv\lib\site-packages\joblib\parallel.py", line 1986, in __call__
    return output if self.return_generator else list(output)
  File "C:\Users\miche\Desktop\projects\moabb\.venv\lib\site-packages\joblib\parallel.py", line 1914, in _get_sequential_output
    res = func(*args, **kwargs)
  File "C:\Users\miche\Desktop\projects\moabb\moabb\evaluations\base.py", line 231, in <lambda>
    lambda d, p: list(
  File "C:\Users\miche\Desktop\projects\moabb\moabb\evaluations\evaluations.py", line 417, in evaluate
    yield from self._evaluate(
  File "C:\Users\miche\Desktop\projects\moabb\moabb\evaluations\evaluations.py", line 228, in _evaluate
    cvclf.fit(X_[train], y_[train])
  File "C:\Users\miche\Desktop\projects\moabb\.venv\lib\site-packages\sklearn\base.py", line 1365, in wrapper
    return fit_method(estimator, *args, **kwargs)
  File "C:\Users\miche\Desktop\projects\moabb\.venv\lib\site-packages\sklearn\pipeline.py", line 655, in fit
    Xt = self._fit(X, y, routed_params, raw_params=params)
  File "C:\Users\miche\Desktop\projects\moabb\.venv\lib\site-packages\sklearn\pipeline.py", line 589, in _fit
    X, fitted_transformer = fit_transform_one_cached(
  File "C:\Users\miche\Desktop\projects\moabb\.venv\lib\site-packages\joblib\memory.py", line 326, in __call__
    return self.func(*args, **kwargs)
  File "C:\Users\miche\Desktop\projects\moabb\.venv\lib\site-packages\sklearn\pipeline.py", line 1540, in _fit_transform_one
    res = transformer.fit_transform(X, y, **params.get("fit_transform", {}))
  File "C:\Users\miche\Desktop\projects\moabb\.venv\lib\site-packages\sklearn\utils\_set_output.py", line 316, in wrapped
    data_to_wrap = f(self, X, *args, **kwargs)
  File "C:\Users\miche\Desktop\projects\moabb\.venv\lib\site-packages\pyriemann\estimation.py", line 402, in fit_transform
    return self.fit(X, y).transform(X)
  File "C:\Users\miche\Desktop\projects\moabb\.venv\lib\site-packages\pyriemann\estimation.py", line 353, in fit
    self.Xd_.fit(X, y)
  File "C:\Users\miche\Desktop\projects\moabb\.venv\lib\site-packages\pyriemann\spatialfilters.py", line 123, in fit
    evals, evecs = eigh(C, Cx)
  File "C:\Users\miche\Desktop\projects\moabb\.venv\lib\site-packages\scipy\linalg\_decomp.py", line 458, in eigh
    a1 = _asarray_validated(a, check_finite=check_finite)
  File "C:\Users\miche\Desktop\projects\moabb\.venv\lib\site-packages\scipy\_lib\_util.py", line 537, in _asarray_validated
    a = toarray(a)
  File "C:\Users\miche\Desktop\projects\moabb\.venv\lib\site-packages\numpy\lib\_function_base_impl.py", line 646, in asarray_chkfinite
    raise ValueError(
ValueError: array must not contain infs or NaNs
```

BRomans replied:
I solved all the issues and updated the docs files.

bruAristimunha commented on moabb/datasets/romani_bf2025_erp.py:
```diff
@@ -0,0 +1,648 @@
+
+    [1] M. Romani, D. Zanoni, E. Farella, and L. Turchet, “BrainForm: a Serious Game for BCI Training and Data Collection,”
+    Oct. 14, 2025, arXiv: arXiv:2510.10169. doi: 10.48550/arXiv.2510.10169.
+
```
```suggestion
```

bruAristimunha commented on moabb/datasets/romani_bf2025_erp.py:
```diff
@@ -0,0 +1,648 @@
+            with zipfile.ZipFile(zip_path, "r") as zip_ref:
+                zip_ref.extractall(cache_dir)
+
+            print("Extraction complete!")
```
```suggestion
```

bruAristimunha commented on moabb/datasets/romani_bf2025_erp.py:
```diff
@@ -0,0 +1,648 @@
+                subjects = self._discover_subjects()
+                return len(subjects) > 0
+            except Exception:
+                return False
```
```suggestion
```

bruAristimunha commented on moabb/datasets/romani_bf2025_erp.py:
```diff
@@ -0,0 +1,648 @@
+
+        print(
+            f"\nMapping original event IDs {original_ids} to Target={t_target} and NonTarget={nt_target}"
+        )
```
```suggestion
        print(
            f"\nMapping original event IDs {original_ids} to Target={t_target} and NonTarget={nt_target}"
        )
```

logger here

bruAristimunha commented:
The download function needs to be refactored, here is one example of how to use the mne functions:

https://github.com/braindecode/braindecode/blob/69840f3ed1227ad041bc67b3beebe5b76c384f58/braindecode/datasets/chb_mit.py

In this example, it's doing the same thing as you: downloading a zip file and unzipping it. What you need to configure is the link, the code, etc.

I'm giving you permission to run the CI; you can commit. If you encounter any errors, minor download issues, or anything unrelated to what you've been working on, you can ignore them.

bruAristimunha reviewed (changes requested):
I made my first comments :)

gcattan reviewed (changes requested):
@BRomans were you able to run your example?
Please share in the thread the within session results.

BRomans commented:
Hey @gcattan, I implemented the changes you requested and fixed a couple of bugs with the download.
This is the output of the function call.

The warning `WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.` if a false positive because I convert the annotations internally but then I need to remove the trigger channel to prevent MNE from automatically overwriting them.

`2025-11-13 13:56:12,701 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:13,237 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:13,270 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:56:13,272 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:56:15,318 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:15,700 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:16,096 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:16,130 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:56:16,132 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:56:16,134 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:   0%|          | 0/22 [00:00<?, ?it/s]2025-11-13 13:56:19,100 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:19,726 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:19,758 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:56:19,760 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:   5%|▍         | 1/22 [00:07<02:34,  7.38s/it]2025-11-13 13:56:26,518 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:27,005 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:27,402 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:27,428 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:56:27,430 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:56:27,432 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:   9%|▉         | 2/22 [00:17<03:04,  9.22s/it]2025-11-13 13:56:36,828 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:37,353 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:37,712 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:37,739 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:56:37,742 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:56:37,744 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  14%|█▎        | 3/22 [00:28<03:05,  9.79s/it]2025-11-13 13:56:47,548 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:48,525 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:48,990 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:49,022 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:56:49,024 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:56:49,029 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  18%|█▊        | 4/22 [00:39<03:05, 10.30s/it]2025-11-13 13:56:58,550 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:59,092 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:59,555 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:56:59,588 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:56:59,590 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:56:59,591 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  23%|██▎       | 5/22 [00:50<02:59, 10.54s/it]2025-11-13 13:57:09,552 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:57:10,395 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:57:10,426 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:57:10,428 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  27%|██▋       | 6/22 [00:58<02:34,  9.65s/it]2025-11-13 13:57:17,395 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:57:18,185 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:57:18,671 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:57:18,699 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:57:18,701 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:57:18,703 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  32%|███▏      | 7/22 [01:09<02:32, 10.15s/it]2025-11-13 13:57:28,496 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:57:29,239 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:57:29,596 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:57:29,631 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:57:29,633 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:57:29,635 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  36%|███▋      | 8/22 [01:20<02:25, 10.38s/it]2025-11-13 13:57:39,281 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:57:39,878 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:57:40,264 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:57:40,290 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:57:40,292 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:57:40,293 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  41%|████      | 9/22 [01:30<02:14, 10.32s/it]2025-11-13 13:57:49,548 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:57:49,953 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:57:50,293 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:57:50,326 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:57:50,328 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:57:50,329 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  45%|████▌     | 10/22 [01:40<02:03, 10.29s/it]2025-11-13 13:57:59,879 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:00,461 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:00,861 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:00,887 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:58:00,889 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:58:00,891 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  50%|█████     | 11/22 [01:51<01:54, 10.43s/it]2025-11-13 13:58:10,562 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:11,069 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:11,540 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:11,574 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:58:11,576 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:58:11,578 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  55%|█████▍    | 12/22 [02:01<01:44, 10.40s/it]2025-11-13 13:58:21,018 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:21,776 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:21,801 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:58:21,803 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  59%|█████▉    | 13/22 [02:09<01:26,  9.59s/it]2025-11-13 13:58:28,657 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:29,357 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:29,392 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:58:29,394 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  64%|██████▎   | 14/22 [02:17<01:11,  8.98s/it]2025-11-13 13:58:36,175 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:36,705 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:37,012 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:37,040 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:58:37,042 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:58:37,045 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  68%|██████▊   | 15/22 [02:27<01:05,  9.36s/it]2025-11-13 13:58:46,444 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:47,023 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:47,334 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:47,353 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:58:47,355 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:58:47,357 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  73%|███████▎  | 16/22 [02:37<00:57,  9.65s/it]2025-11-13 13:58:56,688 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:57,220 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:57,612 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:58:57,645 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:58:57,647 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:58:57,648 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  77%|███████▋  | 17/22 [02:48<00:49,  9.99s/it]2025-11-13 13:59:07,494 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:59:08,134 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:59:08,450 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:59:08,475 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:59:08,477 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:59:08,478 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  82%|████████▏ | 18/22 [02:58<00:39,  9.98s/it]2025-11-13 13:59:17,598 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:59:18,451 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:59:18,488 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:59:18,490 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  86%|████████▋ | 19/22 [03:05<00:27,  9.15s/it]2025-11-13 13:59:24,716 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:59:25,445 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:59:25,478 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:59:25,479 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  91%|█████████ | 20/22 [03:12<00:17,  8.58s/it]2025-11-13 13:59:32,015 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:59:32,635 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:59:33,077 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:59:33,101 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:59:33,103 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:59:33,105 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession:  95%|█████████▌| 21/22 [03:23<00:09,  9.29s/it]2025-11-13 13:59:42,887 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:59:43,346 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:59:43,750 INFO MainThread root Mapping original event IDs [ 1  2  3  4  5  6  7  8  9 10] to Target=1 and NonTarget=2
2025-11-13 13:59:43,772 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:59:43,774 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
2025-11-13 13:59:43,775 WARNING MainThread moabb.datasets.preprocessing No stim channel nor annotations found, skipping setting annotations.
RomaniBF2025ERP-WithinSession: 100%|██████████| 22/22 [03:34<00:00,  9.73s/it]
2025-11-13 13:59:52,328 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P01 | 0grain: Score 0.964
2025-11-13 13:59:52,515 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P01 | 0grain: Score 0.984
2025-11-13 13:59:52,726 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P01 | 1cb: Score 0.978
2025-11-13 13:59:52,892 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P01 | 1cb: Score 0.986
2025-11-13 13:59:53,100 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P02 | 0cb: Score 0.950
2025-11-13 13:59:53,321 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P02 | 0cb: Score 0.977
2025-11-13 13:59:53,541 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P02 | 1grain: Score 0.964
2025-11-13 13:59:53,729 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P02 | 1grain: Score 0.974
2025-11-13 13:59:53,908 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P02 | 2grainExtra: Score 0.847
2025-11-13 13:59:54,108 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P02 | 2grainExtra: Score 0.887
2025-11-13 13:59:54,325 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P03 | 0grain: Score 0.988
2025-11-13 13:59:54,517 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P03 | 0grain: Score 0.994
2025-11-13 13:59:54,722 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P03 | 1cb: Score 0.992
2025-11-13 13:59:54,925 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P03 | 1cb: Score 0.994
2025-11-13 13:59:55,144 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P03 | 2cbExtra: Score 0.994
2025-11-13 13:59:55,333 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P03 | 2cbExtra: Score 0.996
2025-11-13 13:59:55,523 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P04 | 0cb: Score 0.941
2025-11-13 13:59:55,716 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P04 | 0cb: Score 0.949
2025-11-13 13:59:55,939 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P04 | 1grain: Score 0.983
2025-11-13 13:59:56,148 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P04 | 1grain: Score 0.987
2025-11-13 13:59:56,373 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P04 | 2cbExtra: Score 0.978
2025-11-13 13:59:56,593 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P04 | 2cbExtra: Score 0.963
2025-11-13 13:59:56,789 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P05 | 0grain: Score 0.963
2025-11-13 13:59:56,977 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P05 | 0grain: Score 0.970
2025-11-13 13:59:57,187 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P05 | 1cb: Score 0.977
2025-11-13 13:59:57,404 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P05 | 1cb: Score 0.972
2025-11-13 13:59:57,612 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P05 | 2cbExtra: Score 0.986
2025-11-13 13:59:57,800 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P05 | 2cbExtra: Score 0.981
2025-11-13 13:59:58,014 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P06 | 0cb: Score 0.954
2025-11-13 13:59:58,203 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P06 | 0cb: Score 0.965
2025-11-13 13:59:58,415 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P06 | 1grain: Score 0.877
2025-11-13 13:59:58,622 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P06 | 1grain: Score 0.921
2025-11-13 13:59:58,805 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P07 | 0grain: Score 0.914
2025-11-13 13:59:59,011 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P07 | 0grain: Score 0.939
2025-11-13 13:59:59,220 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P07 | 1cb: Score 0.968
2025-11-13 13:59:59,433 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P07 | 1cb: Score 0.978
2025-11-13 13:59:59,635 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P07 | 2cbExtra: Score 0.940
2025-11-13 13:59:59,819 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P07 | 2cbExtra: Score 0.972
2025-11-13 14:00:00,008 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P08 | 0cb: Score 0.975
2025-11-13 14:00:00,230 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P08 | 0cb: Score 0.995
2025-11-13 14:00:00,423 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P08 | 1grain: Score 0.950
2025-11-13 14:00:00,607 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P08 | 1grain: Score 0.957
2025-11-13 14:00:00,806 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P08 | 2cbExtra: Score 0.982
2025-11-13 14:00:00,989 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P08 | 2cbExtra: Score 0.995
2025-11-13 14:00:01,196 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P09 | 0grain: Score 0.964
2025-11-13 14:00:01,397 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P09 | 0grain: Score 0.980
2025-11-13 14:00:01,624 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P09 | 1cb: Score 0.989
2025-11-13 14:00:01,819 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P09 | 1cb: Score 0.989
2025-11-13 14:00:02,023 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P09 | 2grainExtra: Score 0.979
2025-11-13 14:00:02,229 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P09 | 2grainExtra: Score 0.978
2025-11-13 14:00:02,461 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P10 | 0cb: Score 0.981
2025-11-13 14:00:02,669 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P10 | 0cb: Score 0.977
2025-11-13 14:00:02,874 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P10 | 1grain: Score 0.971
2025-11-13 14:00:03,076 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P10 | 1grain: Score 0.969
2025-11-13 14:00:03,278 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P10 | 2grainExtra: Score 0.998
2025-11-13 14:00:03,484 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P10 | 2grainExtra: Score 0.999
2025-11-13 14:00:03,692 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P11 | 0grain: Score 0.944
2025-11-13 14:00:03,878 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P11 | 0grain: Score 0.943
2025-11-13 14:00:04,094 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P11 | 1cb: Score 0.990
2025-11-13 14:00:04,296 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P11 | 1cb: Score 0.993
2025-11-13 14:00:04,516 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P11 | 2grainExtra: Score 0.986
2025-11-13 14:00:04,738 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P11 | 2grainExtra: Score 0.994
2025-11-13 14:00:04,914 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P12 | 0cb: Score 0.954
2025-11-13 14:00:05,129 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P12 | 0cb: Score 0.985
2025-11-13 14:00:05,345 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P12 | 1grain: Score 0.866
2025-11-13 14:00:05,536 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P12 | 1grain: Score 0.933
2025-11-13 14:00:05,740 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P12 | 2cbExtra: Score 0.943
2025-11-13 14:00:05,958 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P12 | 2cbExtra: Score 0.971
2025-11-13 14:00:06,158 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P13 | 0grain: Score 0.923
2025-11-13 14:00:06,365 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P13 | 0grain: Score 0.962
2025-11-13 14:00:06,569 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P13 | 1cb: Score 0.950
2025-11-13 14:00:06,769 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P13 | 1cb: Score 0.964
2025-11-13 14:00:06,975 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P14 | 0cb: Score 0.926
2025-11-13 14:00:07,148 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P14 | 0cb: Score 0.971
2025-11-13 14:00:07,362 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P14 | 1grain: Score 0.974
2025-11-13 14:00:07,580 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P14 | 1grain: Score 0.991
2025-11-13 14:00:07,799 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P16 | 0cb: Score 0.986
2025-11-13 14:00:07,987 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P16 | 0cb: Score 0.993
2025-11-13 14:00:08,171 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P16 | 1grain: Score 0.995
2025-11-13 14:00:08,389 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P16 | 1grain: Score 0.998
2025-11-13 14:00:08,603 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P16 | 2cbExtra: Score 1.000
2025-11-13 14:00:08,799 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P16 | 2cbExtra: Score 1.000
2025-11-13 14:00:08,997 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P17 | 0grain: Score 0.888
2025-11-13 14:00:09,216 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P17 | 0grain: Score 0.912
2025-11-13 14:00:09,410 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P17 | 1cb: Score 0.983
2025-11-13 14:00:09,612 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P17 | 1cb: Score 0.984
2025-11-13 14:00:09,801 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P17 | 2cbExtra: Score 0.975
2025-11-13 14:00:10,009 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P17 | 2cbExtra: Score 0.981
2025-11-13 14:00:10,227 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P19 | 0grain: Score 0.984
2025-11-13 14:00:10,436 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P19 | 0grain: Score 0.987
2025-11-13 14:00:10,635 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P19 | 1cb: Score 0.994
2025-11-13 14:00:10,827 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P19 | 1cb: Score 0.995
2025-11-13 14:00:11,045 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P19 | 2cbExtra: Score 0.990
2025-11-13 14:00:11,232 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P19 | 2cbExtra: Score 0.992
2025-11-13 14:00:11,438 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P20 | 0cb: Score 0.986
2025-11-13 14:00:11,660 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P20 | 0cb: Score 0.982
2025-11-13 14:00:11,867 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P20 | 1grain: Score 0.983
2025-11-13 14:00:12,086 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P20 | 1grain: Score 1.000
2025-11-13 14:00:12,295 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P20 | 2grainExtra: Score 0.968
2025-11-13 14:00:12,513 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P20 | 2grainExtra: Score 0.994
2025-11-13 14:00:12,733 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P21 | 0grain: Score 0.944
2025-11-13 14:00:12,938 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P21 | 0grain: Score 0.972
2025-11-13 14:00:13,128 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P21 | 1cb: Score 0.977
2025-11-13 14:00:13,333 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P21 | 1cb: Score 0.982
2025-11-13 14:00:13,548 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P22 | 0cb: Score 0.973
2025-11-13 14:00:13,748 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P22 | 0cb: Score 0.989
2025-11-13 14:00:13,961 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P22 | 1grain: Score 0.991
2025-11-13 14:00:14,177 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P22 | 1grain: Score 0.997
2025-11-13 14:00:14,388 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P23 | 0grain: Score 0.901
2025-11-13 14:00:14,600 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P23 | 0grain: Score 0.933
2025-11-13 14:00:14,824 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P23 | 1cb: Score 0.953
2025-11-13 14:00:15,035 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P23 | 1cb: Score 0.975
2025-11-13 14:00:15,246 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P23 | 2cbExtra: Score 0.922
2025-11-13 14:00:15,451 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P23 | 2cbExtra: Score 0.958
2025-11-13 14:00:15,659 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P24 | 0cb: Score 0.968
2025-11-13 14:00:15,825 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P24 | 0cb: Score 0.974
2025-11-13 14:00:16,044 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P24 | 1grain: Score 0.993
2025-11-13 14:00:16,251 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P24 | 1grain: Score 0.993
2025-11-13 14:00:16,466 INFO MainThread moabb.evaluations.base RG+LDA | RomaniBF2025ERP | P24 | 2grainExtra: Score 0.996
2025-11-13 14:00:16,681 INFO MainThread moabb.evaluations.base Xdw+LDA | RomaniBF2025ERP | P24 | 2grainExtra: Score 0.997`

gcattan commented on moabb/datasets/romani_bf2025_erp.py:
```diff
@@ -0,0 +1,489 @@
+            doi="10.48550/arXiv.2510.10169",
+        )
+
+    def _download_and_extract_dataset(self) -> str:
```
I am not sure. Why do we need this one?
MOABB should be doing it for you already.

BRomans replied:
Because for some reason it creates a folder named "BIDS.unzipped" instead of the folder name provided as parameter "BrainForm-BIDS-eeg-dataset". So I needed to fix it manually.

gcattan commented on moabb/datasets/romani_bf2025_erp.py:
```diff
@@ -0,0 +1,489 @@
+            raise FileNotFoundError(f"Dataset folder not found: {self.data_folder}")
+        return self.data_folder
+
+    def data_url(
```
It should be a private method.

gcattan commented on moabb/datasets/romani_bf2025_erp.py:
```diff
@@ -0,0 +1,489 @@
+
+        return subjects
+
+    def get_session_list(self, subject):
```
I don t think we need this. Can you remove please?

BRomans commented:
@gcattan I understand your concern.
I am able to indeed suppress the warning by adding a filter in my dataset class
`moabb_logger = logging.getLogger('moabb.datasets.preprocessing')

class IgnoreNoStimWarning(logging.Filter):
    def filter(self, record):
        return 'No stim channel nor annotations found, skipping setting annotations.' not in record.getMessage()

moabb_logger.addFilter(IgnoreNoStimWarning())`

However, I do not believe it to be a very elegant solution.

The problem lies in this function, called by `make_process_pipelines()`

```
  def transform(self, raw, y=None):
        duration = self.interval[1] - self.interval[0]
        offset = int(self.interval[0] * raw.info["sfreq"])
        stim_channels = mne.utils._get_stim_channel(None, raw.info, raise_error=False)
        if len(stim_channels) == 0:
            log.warning(
                "No stim channel nor annotations found, skipping setting annotations."
            )
            return raw
        events = mne.find_events(raw, shortest_event=0, verbose=False)
        events = _unsafe_pick_events(events, include=_get_event_id_values(self.event_id))
        events[:, 0] += offset
        if len(events) != 0:
            annotations = mne.annotations_from_events(
                events,
                raw.info["sfreq"],
                self.event_desc,
                first_samp=raw.first_samp,
                verbose=False,
            )
            annotations.set_durations(duration)
            raw.set_annotations(annotations)
        else:
            log.warning("No events found, skipping setting annotations.")
        return raw
```

Basically, it assumes that annotations must be recreated from stim channel. Thus, even if I update the values of the triggers to match the convetion 1:"Target" and 2:"NonTarget", the newly created annotations do no respect the naming and the pipeline crashes.
You tell me what you think is best: warning suppression is easy and I can push it immediately, otherwise I can patch these functions to pick events from annotations, if present.

gcattan commented:
@BRomans sorry I don t get it. 
You are using `mne.events_from_annotations` to get events.
Can you recreate the stimulus channel out of it, and just add it to the raw data?
